src/custom-lists/background/storage.ts

```diff
@@ -0,0 +1,191 @@
+            }
+        })
+        return mappedLists
+    }
+
+    async fetchListById(id: Number) {
+        const list = await this.storageManager.findObject(COLLECTION_NAME, { id })
+        const pages = await this.storageManager.findAll(PAGE_LIST_ENTRY, { listId: list.id })
+        delete list["_name_terms"]
+        return {
+            ...list,
+            pages: pages.map((page: PageObject) => page.fullUrl),
+        }
+    }
+
+    // Return all the pages associated with a list.
+    async fetchListPages(listId: Number) {
+        return await this.storageManager.findAll(PAGE_LIST_ENTRY, { listId })
+    }
+
+    // Returns all the lists containing a certain page.
+    async getListAssocPage({ url }: { url: string }) {
+        const pages = await this.storageManager.findAll(PAGE_LIST_ENTRY, {
+            pageUrl: url,
+        })
+        const listIds = pages.map(({ listId }: PageObject) => listId)
+        const lists = await this.fetchAllList({
+            query: {
+                id: { $in: listIds }
+            }
+        })
+        return this.changeListsbeforeSending(lists, pages)
+    }
+
+    // Function to insert into the DB
+    async insertCustomList({ name, isDeletable = 1, isNestable = 1 }: {
+        name: string, isDeletable: 0 | 1, isNestable: 0 | 1
+    }) {
+        return await this.storageManager.putObject(COLLECTION_NAME, {
+            name,
+            isDeletable,
+            isNestable,
+            createdAt: new Date(),
+        })
+    }
+
+    async updateListName({ id, name }: { id: Number, name: string }) {
+        await this.storageManager.updateObject(COLLECTION_NAME, {
+            id
+        },
+            {
+                $set: {
+                    name,
+                    createdAt: new Date(),
+                }
+            })
+    }
+
+    // Delete List from the DB.
+    async removeList({ id }: { id: number }) {
+        await this.storageManager.deleteObject(COLLECTION_NAME, {
+            id,
+        })
+        // Delete All pages associated with that list also
+        await this.storageManager.deleteObject(PAGE_LIST_ENTRY, {
+            listId: id,
+        })
+    }
+
+    // Adds mapping to lists and pages table.
+    async insertPageToList({ listId, pageUrl, fullUrl }:
+        { listId: number, pageUrl: string, fullUrl: string }) {
+
+        // check if the list ID exists in the DB, if not cannot add.
+        const idExists = Boolean(await this.fetchListById(listId))
+
+        if (idExists) {
+            await this.storageManager.putObject(PAGE_LIST_ENTRY, {
+                listId,
+                pageUrl,
+                fullUrl,
+                createdAt: new Date(),
+            })
+        }
+    }
+
+    // Removes the page from list.
+    async removePageFromList({ listId, pageUrl }: {
+        listId: number, pageUrl: number
+    }) {
+        const x = await this.storageManager.deleteObject(PAGE_LIST_ENTRY, {
+            $and: [
```

Comment:
Do you know what's the difference from doing the query this way to just doing it like `{ listId, pageUrl }`? Does both work?

I will check the source for dexie-mongify at some point and see what each are translated as (I'm mainly worried about it translating it into multiple queries because of the `$and` rather than doing a single compound index query, like how the query in `getListNameSuggestions` works).